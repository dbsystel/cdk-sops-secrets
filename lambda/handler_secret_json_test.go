package main

import (
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
)

func Test_FullWorkflow_Create_S3_JSON_Simple(t *testing.T) {
	mocks, ctx, event := prepareHandler(t, "events/event_create_s3_secret_json_simple.json")

	phys, data, err := mocks.syncSopsToSecretsmanager(ctx, event)
	check(err)
	snaps.MatchSnapshot(t, ">>>syncSopsToSecretsmanager", phys, data, err)
}

func Test_FullWorkflow_Create_S3_JSON_Complex(t *testing.T) {
	mocks, ctx, event := prepareHandler(t, "events/event_create_s3_secret_json_complex.json")

	phys, data, err := mocks.syncSopsToSecretsmanager(ctx, event)
	check(err)
	snaps.MatchSnapshot(t, ">>>syncSopsToSecretsmanager", phys, data, err)
}

func Test_FullWorkflow_Create_S3_JSON_Complex_StringifyValues(t *testing.T) {
	mocks, ctx, event := prepareHandler(t, "events/event_create_s3_secret_json_complex_stringify.json")

	phys, data, err := mocks.syncSopsToSecretsmanager(ctx, event)
	check(err)
	snaps.MatchSnapshot(t, ">>>syncSopsToSecretsmanager", phys, data, err)
}
func Test_FullWorkflow_Create_S3_JSON_Complex_Flat(t *testing.T) {
	mocks, ctx, event := prepareHandler(t, "events/event_create_s3_secret_json_complex_flat.json")

	phys, data, err := mocks.syncSopsToSecretsmanager(ctx, event)
	check(err)
	snaps.MatchSnapshot(t, ">>>syncSopsToSecretsmanager", phys, data, err)
}

func Test_FullWorkflow_Create_INLINE_JSON_Simple(t *testing.T) {
	mocks, ctx, event := prepareHandler(t, "events/event_create_s3_secret_json_simple.json")
	event = fileToInline(event)

	phys, data, err := mocks.syncSopsToSecretsmanager(ctx, event)
	check(err)
	snaps.MatchSnapshot(t, ">>>syncSopsToSecretsmanager", phys, data, err)
}

func Test_FullWorkflow_Create_INLINE_JSON_Complex(t *testing.T) {
	mocks, ctx, event := prepareHandler(t, "events/event_create_s3_secret_json_complex.json")
	event = fileToInline(event)

	phys, data, err := mocks.syncSopsToSecretsmanager(ctx, event)
	check(err)
	snaps.MatchSnapshot(t, ">>>syncSopsToSecretsmanager", phys, data, err)
}

func Test_FullWorkflow_Create_INLINE_JSON_Complex_StringifyValues(t *testing.T) {
	mocks, ctx, event := prepareHandler(t, "events/event_create_s3_secret_json_complex_stringify.json")
	event = fileToInline(event)

	phys, data, err := mocks.syncSopsToSecretsmanager(ctx, event)
	check(err)
	snaps.MatchSnapshot(t, ">>>syncSopsToSecretsmanager", phys, data, err)
}
func Test_FullWorkflow_Create_INLINE_JSON_Complex_Flat(t *testing.T) {
	mocks, ctx, event := prepareHandler(t, "events/event_create_s3_secret_json_complex_flat.json")
	event = fileToInline(event)

	phys, data, err := mocks.syncSopsToSecretsmanager(ctx, event)
	check(err)
	snaps.MatchSnapshot(t, ">>>syncSopsToSecretsmanager", phys, data, err)
}
