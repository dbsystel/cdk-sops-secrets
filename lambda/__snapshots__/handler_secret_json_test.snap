
[Test_FullWorkflow_Create_S3_JSON_Simple - 1]
>>>S3ApiMockClient.GetObjectAttributes.Input
{
  Bucket: "..",
  Key: "../test-secrets/json/sopsfile.enc-age.json"
}
---

[Test_FullWorkflow_Create_S3_JSON_Simple - 2]
>>>S3MAnagerMockClient.Download.Input
{
  Bucket: "..",
  Key: "../test-secrets/json/sopsfile.enc-age.json"
}
---

[Test_FullWorkflow_Create_S3_JSON_Simple - 3]
>>>SecretsManagerMockClient.PutSecretValue.Input
main.putSecretValueInputNotSecure{
    _:                  struct {}{},
    ClientRequestToken: &"some-32-character-long-string-ab",
    RotationToken:      (*string)(nil),
    SecretBinary:       nil,
    SecretId:           &"arn:aws:secretsmanager:eu-central-1:123456789012:secret:testsecret",
    SecretString:       &"{\n  \"key1\": \"value1\",\n  \"key2\": 12345,\n  \"key3\": false\n}",
    VersionStages:      nil,
}
---

[Test_FullWorkflow_Create_S3_JSON_Simple - 4]
>>>syncSopsToSecretsmanager
arn:custom:sopssync:eu-central-1:123456789012:secret:testsecret
map[string]interface {}{
    "ARN":           "arn:custom:sopssync:eu-central-1:123456789012:secret:testsecret",
    "Name":          "dfb70680d326c245dbdd113d635caa16113e78b1eb8b7447f10eeb861e491724",
    "VersionId":     "46b116fd9409581eb2623320d877cfa1f46cb2487bc3cd0672b1eb95554e65d3",
    "VersionStages": []*string{
        &"dfb70680d326c245dbdd113d635caa16113e78b1eb8b7447f10eeb861e491724",
    },
}
nil
---

[Test_FullWorkflow_Create_S3_JSON_Complex - 1]
>>>S3ApiMockClient.GetObjectAttributes.Input
{
  Bucket: "..",
  Key: "../test-secrets/json/sopsfile-complex.enc-age.json"
}
---

[Test_FullWorkflow_Create_S3_JSON_Complex - 2]
>>>S3MAnagerMockClient.Download.Input
{
  Bucket: "..",
  Key: "../test-secrets/json/sopsfile-complex.enc-age.json"
}
---

[Test_FullWorkflow_Create_S3_JSON_Complex - 3]
>>>SecretsManagerMockClient.PutSecretValue.Input
main.putSecretValueInputNotSecure{
    _:                  struct {}{},
    ClientRequestToken: &"some-32-character-long-string-ab",
    RotationToken:      (*string)(nil),
    SecretBinary:       nil,
    SecretId:           &"arn:aws:secretsmanager:eu-central-1:123456789012:secret:testsecret",
    SecretString:       &"{\n  \"and now\": {\n    \"some\": [\n      {\n        \"basic\": false\n      },\n      {\n        \"nested\": 12345\n      },\n      {\n        \"type\": 1.2345\n      },\n      {\n        \"tests\": \"Finish!\"\n      }\n    ]\n  },\n  \"some\": {\n    \"deep\": {\n      \"nested\": {\n        \"arrays\": [\n          \"with\",\n          \"several\",\n          {\n            \"values\": {\n              \"and\": \"objects\"\n            }\n          }\n        ],\n        \"object\": \"structure\"\n      }\n    },\n    \"notsodeep\": \"struct\"\n  }\n}",
    VersionStages:      nil,
}
---

[Test_FullWorkflow_Create_S3_JSON_Complex - 4]
>>>syncSopsToSecretsmanager
arn:custom:sopssync:eu-central-1:123456789012:secret:testsecret
map[string]interface {}{
    "ARN":           "arn:custom:sopssync:eu-central-1:123456789012:secret:testsecret",
    "Name":          "dfb70680d326c245dbdd113d635caa16113e78b1eb8b7447f10eeb861e491724",
    "VersionId":     "28e49c30b2d8af77e323b63c5c9e90ec2cf1385ab330eb674695ee065b1252f2",
    "VersionStages": []*string{
        &"dfb70680d326c245dbdd113d635caa16113e78b1eb8b7447f10eeb861e491724",
    },
}
nil
---

[Test_FullWorkflow_Create_S3_JSON_Complex_StringifyValues - 1]
>>>S3ApiMockClient.GetObjectAttributes.Input
{
  Bucket: "..",
  Key: "../test-secrets/json/sopsfile-complex.enc-age.json"
}
---

[Test_FullWorkflow_Create_S3_JSON_Complex_StringifyValues - 2]
>>>S3MAnagerMockClient.Download.Input
{
  Bucket: "..",
  Key: "../test-secrets/json/sopsfile-complex.enc-age.json"
}
---

[Test_FullWorkflow_Create_S3_JSON_Complex_StringifyValues - 3]
>>>SecretsManagerMockClient.PutSecretValue.Input
main.putSecretValueInputNotSecure{
    _:                  struct {}{},
    ClientRequestToken: &"some-32-character-long-string-ab",
    RotationToken:      (*string)(nil),
    SecretBinary:       nil,
    SecretId:           &"arn:aws:secretsmanager:eu-central-1:123456789012:secret:testsecret",
    SecretString:       &"{\n  \"and now\": {\n    \"some\": [\n      {\n        \"basic\": \"false\"\n      },\n      {\n        \"nested\": \"12345\"\n      },\n      {\n        \"type\": \"1.2345\"\n      },\n      {\n        \"tests\": \"Finish!\"\n      }\n    ]\n  },\n  \"some\": {\n    \"deep\": {\n      \"nested\": {\n        \"arrays\": [\n          \"with\",\n          \"several\",\n          {\n            \"values\": {\n              \"and\": \"objects\"\n            }\n          }\n        ],\n        \"object\": \"structure\"\n      }\n    },\n    \"notsodeep\": \"struct\"\n  }\n}",
    VersionStages:      nil,
}
---

[Test_FullWorkflow_Create_S3_JSON_Complex_StringifyValues - 4]
>>>syncSopsToSecretsmanager
arn:custom:sopssync:eu-central-1:123456789012:secret:testsecret
map[string]interface {}{
    "ARN":           "arn:custom:sopssync:eu-central-1:123456789012:secret:testsecret",
    "Name":          "dfb70680d326c245dbdd113d635caa16113e78b1eb8b7447f10eeb861e491724",
    "VersionId":     "ddc702937cbed1b68f84659094f8db7c862114642ae2a98baa43c42e16b256f4",
    "VersionStages": []*string{
        &"dfb70680d326c245dbdd113d635caa16113e78b1eb8b7447f10eeb861e491724",
    },
}
nil
---

[Test_FullWorkflow_Create_S3_JSON_Complex_Flat - 1]
>>>S3ApiMockClient.GetObjectAttributes.Input
{
  Bucket: "..",
  Key: "../test-secrets/json/sopsfile-complex.enc-age.json"
}
---

[Test_FullWorkflow_Create_S3_JSON_Complex_Flat - 2]
>>>S3MAnagerMockClient.Download.Input
{
  Bucket: "..",
  Key: "../test-secrets/json/sopsfile-complex.enc-age.json"
}
---

[Test_FullWorkflow_Create_S3_JSON_Complex_Flat - 3]
>>>SecretsManagerMockClient.PutSecretValue.Input
main.putSecretValueInputNotSecure{
    _:                  struct {}{},
    ClientRequestToken: &"some-32-character-long-string-ab",
    RotationToken:      (*string)(nil),
    SecretBinary:       nil,
    SecretId:           &"arn:aws:secretsmanager:eu-central-1:123456789012:secret:testsecret",
    SecretString:       &"{\n  \"and now.some[0].basic\": false,\n  \"and now.some[1].nested\": 12345,\n  \"and now.some[2].type\": 1.2345,\n  \"and now.some[3].tests\": \"Finish!\",\n  \"some.deep.nested.arrays[0]\": \"with\",\n  \"some.deep.nested.arrays[1]\": \"several\",\n  \"some.deep.nested.arrays[2].values.and\": \"objects\",\n  \"some.deep.nested.object\": \"structure\",\n  \"some.notsodeep\": \"struct\"\n}",
    VersionStages:      nil,
}
---

[Test_FullWorkflow_Create_S3_JSON_Complex_Flat - 4]
>>>syncSopsToSecretsmanager
arn:custom:sopssync:eu-central-1:123456789012:secret:testsecret
map[string]interface {}{
    "ARN":           "arn:custom:sopssync:eu-central-1:123456789012:secret:testsecret",
    "Name":          "dfb70680d326c245dbdd113d635caa16113e78b1eb8b7447f10eeb861e491724",
    "VersionId":     "e9a8d3ab5715c1120faae25b669e5905a92fe4b261ae97eb1e65ad9ba52f9ee6",
    "VersionStages": []*string{
        &"dfb70680d326c245dbdd113d635caa16113e78b1eb8b7447f10eeb861e491724",
    },
}
nil
---

[Test_FullWorkflow_Create_INLINE_JSON_Simple - 1]
>>>SecretsManagerMockClient.PutSecretValue.Input
main.putSecretValueInputNotSecure{
    _:                  struct {}{},
    ClientRequestToken: &"MyHash",
    RotationToken:      (*string)(nil),
    SecretBinary:       nil,
    SecretId:           &"arn:aws:secretsmanager:eu-central-1:123456789012:secret:testsecret",
    SecretString:       &"{\n  \"key1\": \"value1\",\n  \"key2\": 12345,\n  \"key3\": false\n}",
    VersionStages:      nil,
}
---

[Test_FullWorkflow_Create_INLINE_JSON_Simple - 2]
>>>syncSopsToSecretsmanager
arn:custom:sopssync:eu-central-1:123456789012:secret:testsecret
map[string]interface {}{
    "ARN":           "arn:custom:sopssync:eu-central-1:123456789012:secret:testsecret",
    "Name":          "dfb70680d326c245dbdd113d635caa16113e78b1eb8b7447f10eeb861e491724",
    "VersionId":     "46b116fd9409581eb2623320d877cfa1f46cb2487bc3cd0672b1eb95554e65d3",
    "VersionStages": []*string{
        &"dfb70680d326c245dbdd113d635caa16113e78b1eb8b7447f10eeb861e491724",
    },
}
nil
---

[Test_FullWorkflow_Create_INLINE_JSON_Complex - 1]
>>>SecretsManagerMockClient.PutSecretValue.Input
main.putSecretValueInputNotSecure{
    _:                  struct {}{},
    ClientRequestToken: &"MyHash",
    RotationToken:      (*string)(nil),
    SecretBinary:       nil,
    SecretId:           &"arn:aws:secretsmanager:eu-central-1:123456789012:secret:testsecret",
    SecretString:       &"{\n  \"and now\": {\n    \"some\": [\n      {\n        \"basic\": false\n      },\n      {\n        \"nested\": 12345\n      },\n      {\n        \"type\": 1.2345\n      },\n      {\n        \"tests\": \"Finish!\"\n      }\n    ]\n  },\n  \"some\": {\n    \"deep\": {\n      \"nested\": {\n        \"arrays\": [\n          \"with\",\n          \"several\",\n          {\n            \"values\": {\n              \"and\": \"objects\"\n            }\n          }\n        ],\n        \"object\": \"structure\"\n      }\n    },\n    \"notsodeep\": \"struct\"\n  }\n}",
    VersionStages:      nil,
}
---

[Test_FullWorkflow_Create_INLINE_JSON_Complex - 2]
>>>syncSopsToSecretsmanager
arn:custom:sopssync:eu-central-1:123456789012:secret:testsecret
map[string]interface {}{
    "ARN":           "arn:custom:sopssync:eu-central-1:123456789012:secret:testsecret",
    "Name":          "dfb70680d326c245dbdd113d635caa16113e78b1eb8b7447f10eeb861e491724",
    "VersionId":     "28e49c30b2d8af77e323b63c5c9e90ec2cf1385ab330eb674695ee065b1252f2",
    "VersionStages": []*string{
        &"dfb70680d326c245dbdd113d635caa16113e78b1eb8b7447f10eeb861e491724",
    },
}
nil
---

[Test_FullWorkflow_Create_INLINE_JSON_Complex_StringifyValues - 1]
>>>SecretsManagerMockClient.PutSecretValue.Input
main.putSecretValueInputNotSecure{
    _:                  struct {}{},
    ClientRequestToken: &"MyHash",
    RotationToken:      (*string)(nil),
    SecretBinary:       nil,
    SecretId:           &"arn:aws:secretsmanager:eu-central-1:123456789012:secret:testsecret",
    SecretString:       &"{\n  \"and now\": {\n    \"some\": [\n      {\n        \"basic\": \"false\"\n      },\n      {\n        \"nested\": \"12345\"\n      },\n      {\n        \"type\": \"1.2345\"\n      },\n      {\n        \"tests\": \"Finish!\"\n      }\n    ]\n  },\n  \"some\": {\n    \"deep\": {\n      \"nested\": {\n        \"arrays\": [\n          \"with\",\n          \"several\",\n          {\n            \"values\": {\n              \"and\": \"objects\"\n            }\n          }\n        ],\n        \"object\": \"structure\"\n      }\n    },\n    \"notsodeep\": \"struct\"\n  }\n}",
    VersionStages:      nil,
}
---

[Test_FullWorkflow_Create_INLINE_JSON_Complex_StringifyValues - 2]
>>>syncSopsToSecretsmanager
arn:custom:sopssync:eu-central-1:123456789012:secret:testsecret
map[string]interface {}{
    "ARN":           "arn:custom:sopssync:eu-central-1:123456789012:secret:testsecret",
    "Name":          "dfb70680d326c245dbdd113d635caa16113e78b1eb8b7447f10eeb861e491724",
    "VersionId":     "ddc702937cbed1b68f84659094f8db7c862114642ae2a98baa43c42e16b256f4",
    "VersionStages": []*string{
        &"dfb70680d326c245dbdd113d635caa16113e78b1eb8b7447f10eeb861e491724",
    },
}
nil
---

[Test_FullWorkflow_Create_INLINE_JSON_Complex_Flat - 1]
>>>SecretsManagerMockClient.PutSecretValue.Input
main.putSecretValueInputNotSecure{
    _:                  struct {}{},
    ClientRequestToken: &"MyHash",
    RotationToken:      (*string)(nil),
    SecretBinary:       nil,
    SecretId:           &"arn:aws:secretsmanager:eu-central-1:123456789012:secret:testsecret",
    SecretString:       &"{\n  \"and now.some[0].basic\": false,\n  \"and now.some[1].nested\": 12345,\n  \"and now.some[2].type\": 1.2345,\n  \"and now.some[3].tests\": \"Finish!\",\n  \"some.deep.nested.arrays[0]\": \"with\",\n  \"some.deep.nested.arrays[1]\": \"several\",\n  \"some.deep.nested.arrays[2].values.and\": \"objects\",\n  \"some.deep.nested.object\": \"structure\",\n  \"some.notsodeep\": \"struct\"\n}",
    VersionStages:      nil,
}
---

[Test_FullWorkflow_Create_INLINE_JSON_Complex_Flat - 2]
>>>syncSopsToSecretsmanager
arn:custom:sopssync:eu-central-1:123456789012:secret:testsecret
map[string]interface {}{
    "ARN":           "arn:custom:sopssync:eu-central-1:123456789012:secret:testsecret",
    "Name":          "dfb70680d326c245dbdd113d635caa16113e78b1eb8b7447f10eeb861e491724",
    "VersionId":     "e9a8d3ab5715c1120faae25b669e5905a92fe4b261ae97eb1e65ad9ba52f9ee6",
    "VersionStages": []*string{
        &"dfb70680d326c245dbdd113d635caa16113e78b1eb8b7447f10eeb861e491724",
    },
}
nil
---
